finetune/optimizers/gradient_accumulation.py

```diff
@@ -44,8 +44,9 @@ def apply_gradients(self, grads_and_vars, global_step=None, name=None, *args, **
 
                 grads_and_accumulated_vars.append((accum_grad, v))
 
-            global_step = global_step if global_step is not None else tf.compat.v1.train.get_or_create_global_step()
-
+            global_step_var = global_step if global_step is not None else tf.compat.v1.train.get_or_create_global_step()
```

Comment:
Can you explain why splitting out into global_step_var and global_step is necessary?